crypto/crypto-derive/src/lib.rs

```diff
@@ -366,12 +366,48 @@ pub fn hasher_dispatch(input: TokenStream) -> TokenStream {
                 self.0.finish()
             }
 
-            fn write(&mut self, bytes: &[u8]) -> &mut Self {
-                self.0.write(bytes);
+            fn update(&mut self, bytes: &[u8]) -> &mut Self {
+                self.0.update(bytes);
                 self
             }
         }
 
+        impl std::io::Write for #hasher_name {
+            fn write(&mut self, bytes: &[u8]) -> std::io::Result<usize> {
+                use libra_crypto::hash::CryptoHasher;
+
+                self.0.update(bytes);
+                Ok(bytes.len())
+            }
+            fn flush(&mut self) -> std::io::Result<()> {
+                Ok(())
+            }
+        }
+
+    );
+    out.into()
+}
+
+#[proc_macro_derive(LCSCryptoHash)]
+pub fn lcs_crypto_hash_dispatch(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+    let name = &ast.ident;
+    let hasher_name = Ident::new(&format!("{}Hasher", &name.to_string()), Span::call_site());
```

Comment:
I take it this call site is the module / struct wherein this is derived?

Reply:
That is part of the macro expansion.